src/main/java/io/molr/gui/fx/widgets/AvailableMissionsView.java

```diff
@@ -75,6 +82,53 @@ public void handle(MouseEvent event) {
             }
         });
     }
+    
+    /**
+     * Maybe encapsulate mole into an auto-reconnect-mole or flux
+     */
+	private void subscribeToMoleStates() {
+
+		Flux<AgencyState> states = mole.states();
+		//clear missions from view in case of connect exceptions. (more sophisticated solutions may be implemented later, like mark them as being offline etc.)
+		states.doOnError(ConnectException.class, this::clearAgencyStatesOnConnectException)
+				.publishOn(FxThreadScheduler.instance())
+				.subscribe(this::update, this::onStatesError, this::onStatesComplete);
+	}
+
+	private void clearAgencyStatesOnConnectException(ConnectException connectException) {
+		LOGGER.info("clear agency states on connect exception", connectException);
+		Platform.runLater(new Runnable() {
+			@Override
+			public void run() {
+				update(new EmptyAgencyStates());
+			}
+		});
+	}
+
+	/*
+	 * Here we could handle all errors completing the flux. The current implementation tries to subscribe again without differentiating any error 
+	 */
+    private void onStatesError(Throwable throwable) {
+    	LOGGER.info("/states subscription has been finished by error", throwable);
```

Comment:
Could this reconnection code be placed in the client class instead of in the panel?